samples/apply-mosaic-rule-to-rasters/src/main/java/com/esri/arcgismaps/sample/applymosaicruletorasters/components/ApplyMosaicRuleToRastersViewModel.kt

```diff
@@ -0,0 +1,192 @@
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.asStateFlow
+import kotlinx.coroutines.launch
+
+/**
+ * Enum representing the available preset mosaic rule types for the sample.
+ */
+enum class MosaicRuleType(val label: String) {
+    ObjectID("Object ID"),
+    NorthWest("North West"),
+    Center("Center"),
+    ByAttribute("By Attribute"),
+    LockRaster("Lock Raster")
+}
+
+/**
+ * ViewModel for the Apply Mosaic Rule to Rasters sample.
+ */
+class ApplyMosaicRuleToRastersViewModel(app: Application) : AndroidViewModel(app) {
+    /**
+     * The ArcGISMap used in the sample, with a topographic basemap.
+     */
+    var arcGISMap by mutableStateOf(
+        ArcGISMap(BasemapStyle.ArcGISTopographic)
+    )
+
+    /**
+     * The ImageServiceRaster from the online raster image service.
+     */
+    private val imageServiceRaster = ImageServiceRaster(IMAGE_SERVICE_URL)
+
+    /**
+     * The RasterLayer that displays the ImageServiceRaster on the map.
+     */
+    private val rasterLayer = RasterLayer(imageServiceRaster)
+
+    /**
+     * The list of available mosaic rule types for the dropdown menu.
+     */
+    val mosaicRuleTypes = MosaicRuleType.entries
+
+    /**
+     * The currently selected mosaic rule type.
+     */
+    private val _selectedRuleType = MutableStateFlow(MosaicRuleType.ObjectID)
+    val selectedRuleType = _selectedRuleType.asStateFlow()
+
+    /**
+     * Loading state for the UI, true while the raster layer is loading.
+     */
+    var isLoading by mutableStateOf(true)
+        private set
+
+    /**
+     * Message dialog for error handling.
+     */
+    val messageDialogVM = MessageDialogViewModel()
+
+    /**
+     * Center point of the raster service (Amberg, Germany), used for recentering the map.
+     */
+    private val imageServiceRasterCenter: Point get() {
+        return imageServiceRaster.serviceInfo?.fullExtent?.extent?.center ?: Point(x = 0.0, y = 0.0)
+    }
+
+    /**
+     * MapViewProxy for controlling the viewpoint after raster is loaded.
+     */
+    val mapViewProxy = MapViewProxy()
+
+    init {
+        // Add the raster layer to the map's operational layers.
+        arcGISMap.operationalLayers.add(rasterLayer)
+        // Set the initial mosaic rule (ObjectID/None).
+        imageServiceRaster.mosaicRule = createMosaicRule(MosaicRuleType.ObjectID)
+        // Load the raster layer and retrieve the center point for future recentering.
+        viewModelScope.launch {
+            rasterLayer.load().onSuccess {
+                // Set the viewpoint to the raster center
+                mapViewProxy.setViewpointAnimated(
+                    Viewpoint(center = imageServiceRasterCenter, scale = 25000.0)
+                )
+                isLoading = false
+            }.onFailure {
+                isLoading = false
+                messageDialogVM.showMessageDialog(it)
+            }
+        }
```

Comment:
This code is duplicated in the function `updateMosaicRule(...)`.

Maybe just call updateMosaicRule here